weatherApp/Features/WeatherDetails/UI/DetailWeatherViewController.swift

```diff
@@ -0,0 +1,146 @@
+//
+//  DetailWeatherViewController.swift
+//  weatherApp
+//
+//  Created by Dario Nikolic on 04/08/2020.
+//  Copyright © 2020 Dario Nikolic. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class DetailWeatherViewController: UIViewController {
+    
+    private var currentWeather: CurrentWeather?
+    private var weatherConditions: [(name: String, value: String)] = []
+    
+    @IBOutlet private weak var tempLabel: UILabel!
+    @IBOutlet private weak var weatherIcon: UIImageView!
+    @IBOutlet private weak var weatherDescriptionLabel: UILabel!
+    @IBOutlet private weak var collectionView: UICollectionView!
+    @IBOutlet private weak var maxTempLabel: UILabel!
+    @IBOutlet private weak var minTempLabel: UILabel!
+    @IBOutlet private weak var collectionViewHeightConstraint: NSLayoutConstraint!
+    
+    convenience init(currentWeather: CurrentWeather) {
+        self.init()
+        
+        self.currentWeather = currentWeather
+    }
+    
+    //MARK: - Overrides
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setWeatherInformation()
+        setupWeatherConditions()
+        setupCollectionView()
+        setupCollectionViewHeight()
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+
+        view.setGradientBackground(startColor: .grayBlueTint, endColor: .darkNavyBlue)
+    }
+    
+    //MARK: CollectioView Data
+    
+    private func setupWeatherConditions() {
+        guard let currentWeather = currentWeather else { return }
+        
+        let forecast = currentWeather.forecast
+        
+        let feelsLikeTemperature = Int(forecast.feelsLikeTemperature.celsius)
+        let windSpeed = currentWeather.wind.speed * 3.6
+        
+        weatherConditions.append((LocalizedStrings.feelsLike, String(format: LocalizedStrings.temperatureValueFormat, feelsLikeTemperature)))
+        weatherConditions.append((LocalizedStrings.humidity,  String(format: LocalizedStrings.percentageValueFormat, forecast.humidity)))
+        weatherConditions.append((LocalizedStrings.pressure,  String(format: LocalizedStrings.pressureValueFormat, forecast.pressure)))
+        weatherConditions.append((LocalizedStrings.wind, String(format: LocalizedStrings.speedValueFormat, windSpeed)))
+    }
+    
+    private func numOfConditions() -> Int {
+        return weatherConditions.count
+    }
+    
+    private func numOfConditionRows() -> Int {
+        return numOfConditions() / 2 + numOfConditions() % 2
+    }
+    
+    //MARK: - UI elements setup
+    
+    private func setWeatherInformation() {
+        guard let currentWeather = currentWeather else { return }
+        
+        let forecast = currentWeather.forecast
+        
+        title = currentWeather.name
+        
+        tempLabel.text = String(String(format: LocalizedStrings.temperatureValueFormat, Int(forecast.temperature.celsius)).dropLast(2))
```

Comment:
This is some magic to build a string :) Please reformat this to some meaningful way. Also, dropFirst without checking might throw an error, check the documentation.